storage/scylla_api.c

```diff
@@ -315,34 +313,51 @@ static status_t get_blob(const CassRow* row, cass_byte_t* target, const char* co
   return ret;
 }
 
-static status_t create_permanent_keyspace(CassSession* session) {
+static char* make_query(const char* head_desc, const char* position, const char* left_desc) {
+  if (head_desc == NULL || position == NULL || left_desc == NULL) {
+    return NULL;
+  }
+  size_t head_len = strlen(head_desc);
+  size_t pos_len = strlen(position);
+  size_t left_len = strlen(left_desc);
+  size_t result_len = head_len + pos_len + left_len + 1;
+  char* result = malloc(result_len);
+
+  memcpy(result, head_desc, head_len);
+  memcpy(result + head_len, position, pos_len);
+  memcpy(result + head_len + pos_len, left_desc, left_len);
+  result[result_len - 1] = 0;
+
+  return result;
+}
+
+static status_t create_permanent_keyspace(CassSession* session, const char* keyspace_name) {
   status_t ret = SC_OK;
-  if (execute_query(session, "DROP KEYSPACE IF EXISTS permanent;") != CASS_OK) {
-    ta_log_error("%s\n", "SC_STORAGE_CASSANDRA_QUREY_FAIL");
-    ret = SC_STORAGE_CASSANDRA_QUREY_FAIL;
-    goto exit;
+  char* create_query = make_query("CREATE KEYSPACE IF NOT EXISTS ", keyspace_name,
+                                  " WITH replication = {"
+                                  "'class': 'SimpleStrategy', 'replication_factor': '2'};");
+  if (create_query == NULL) {
```

Comment:
I think OOM would be a more appropriate error code, since `create_query == NULL` happens when there is no space for `create_quert`

Reply:
I am not sure which to use since both input NULL pointer and oom will return NULL and set to create_query.

Reply:
But in this case actually only OOM would happen.